subsys/fs/fs_automount.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <zephyr/types.h>
+#include <errno.h>
+#include <init.h>
+#include <fs/fs.h>
+#include <fs/littlefs.h>
+#include <sys/stat.h>
+
+#define DT_DRV_COMPAT fixed_partitions
+#include <devicetree.h>
+
+#define LOG_LEVEL CONFIG_FS_LOG_LEVEL
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(fs_automount);
+
+const extern struct flash_area *flash_map;
+const extern int flash_map_entries;
+
+#define LITTLEFS_MNT_DATA(part)						\
+	case DT_FIXED_PARTITION_ID(part):				\
+	{								\
+		FS_LITTLEFS_DECLARE_DEFAULT_CONFIG(fsdata);		\
+		static struct fs_mount_t mntdata = {			\
+			.type = DT_PROP_OR(part, mnttype, FS_DEFAULT),	\
+			.mnt_point = DT_PROP_OR(part, mntpoint, NULL),	\
+		.fs_data = &fsdata,					\
+		.storage_dev = (void *)DT_FIXED_PARTITION_ID(part),	\
+	};								\
+	return &mntdata;						\
+	}
+
+#define DEFINE_PARITION_DATA(part)			\
+	COND_CODE_1(DT_NODE_HAS_PROP(part, mntpoint),	\
+		    (LITTLEFS_MNT_DATA(part)), ())
+
+#define FOREACH_PARTION(n) DT_FOREACH_CHILD(DT_DRV_INST(n), DEFINE_PARITION_DATA)
+
+static struct fs_mount_t *fs_mount_obtain(int id)
+{
+	switch (id) {
+	DT_INST_FOREACH_STATUS_OKAY(FOREACH_PARTION)
+	}
+
+	return NULL;
+}
+
+static int fs_automount(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	int ret = 0;
+
+	for (int i = 0; i < flash_map_entries; ++i) {
+		static struct fs_mount_t *mnt_data;
+
+		mnt_data = fs_mount_obtain(i);
+		if (mnt_data != NULL) {
+			printk("Mount point for index %d is %s\n", i, mnt_data->mnt_point);
+			ret = fs_mount(mnt_data);
+			printk("mount returned %d\n", ret);
+		}
+	}
+
+	return ret;
+}
+
+
+SYS_INIT(fs_automount, POST_KERNEL, CONFIG_APPLICATION_INIT_PRIORITY);
```

Comment:
There's no provision here for identifying device dependencies because we can't tell from the devicetree what the mount attempt requires.  I guess that could be ok, but in the long term dependencies should not rely on human-assigned priorities.